config/routes.rb

```diff
@@ -1,4 +1,9 @@
 Rails.application.routes.draw do
+  get '/users/:user_id/posts', to: 'posts#index', as: 'user_posts'
+  get '/users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
+  get '/users', to: 'users#index'
+  get '/users/:id', to: 'users#show', as: 'user'
+  root 'users#index'
```

Comment:
- [ ] I suggest that you use `resource` routing allows you to quickly declare all of the common routes for a given resourceful controller, you can do:
 ``` 
   resources :users, only: [:index, :show] do
       resources :posts, only: [:index, :show]
   end
```